### ISSUE TITLE
nfs() doesn't add "_" (space) in front of 0

### ISSUE THREAD

**Issue by Nekzuris** [processing/p5.js#3738]
#### Nature of issue?

- [x] Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [x] String Functions: nfs()

#### Which platform were you using when you encountered this?

- [ ] Mobile/Tablet (touch devices)
- [x] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

The nfs() function doens't add a space in front of the value 0.
current behavior:
`nfs(0, 1, 2) = "0.00"`
instead of:
`nfs(0, 1, 2) = "_0.00"`
( "_" represent a space in the above examples)

- p5.js version: 0.8.0
- Web browser and version: Firefox 66.0.5
- Operating System: Windows 10
- Steps to reproduce this:
try `nfs(0, 1, 2)`

**Comment by stalgiag** [comment]
This is a bit ambiguous. The documentation says this method "puts an additional "_" (space) in front of positive numbers" but [this check ](https://github.com/processing/p5.js/blob/master/src/utilities/string_functions.js#L456) will ignore 0. If we want to allow this to put spaces in front of 0 we could do this instead:
```
function addNfs(num) {
  return Math.sign(parseFloat(num)) > -1
    ? ' ' + num.toString()
    : num.toString();
}
```

**Comment by stalgiag** [comment]
Or `parseFloat(num) >= 0` haha

**Comment by Nekzuris** [comment]
> Or `parseFloat(num) >= 0` 

Year that seems better ;)
I really think this should be the correct behavior of nfs(), there is no reason to exclude zero here.

**Comment by stalgiag** [comment]
I agree. Any interest in opening a pull request?

**Comment by Nekzuris** [comment]
Yes, should I do it? I'm just getting into programming, never use github before, not really sure how to do it.

**Comment by stalgiag** [comment]
That seems like a good time to give it a try to me. But also no pressure! This isn't high priority so time can be taken with it.

If you do decide to give it a go. [This can be a useful resource.](https://github.com/processing/p5.js/tree/master/developer_docs).  After you figure out how to build the library in a local copy of your fork (ie run `npm run grunt` and see all of the tests pass etc) then you can make a new branch where you just change [this line](https://github.com/processing/p5.js/blob/master/src/utilities/string_functions.js#L456) to:
```
return parseFloat(num) >= 0 ? ' ' + num.toString() : num.toString();
```
and in the documentation in [this line](https://github.com/processing/p5.js/blob/master/src/utilities/string_functions.js#L385) you can change "positive" to "non-negative". Then if you can still run `npm run grunt` successfully you are free to commit those two changes, push to your fork, and open a PR here with your branch.

Also, if you do decide to give it a go and you need help feel free to ping me with @stalgiag. You can also do that if you are sure that you don't want to do it, and I can add the "help wanted" tag to this issue to enlist help from someone else.

**Comment by Nekzuris** [comment]
Thanks for your help, I will give it a try ;-)

**Comment by Nekzuris** [comment]
@stalgiag I struggle to build from the source with `npm run grunt`, I'm getting these errors:
Aborted due to warnings.
```bash
npm ERR! Linux 4.18.0-14-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "grunt"
npm ERR! node v8.10.0
npm ERR! npm  v3.5.2
npm ERR! code ELIFECYCLE
npm ERR! p5@0.8.0 grunt: `grunt`
npm ERR! Exit status 3
npm ERR! 
npm ERR! Failed at the p5@0.8.0 grunt script 'grunt'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the p5 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     grunt
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs p5
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls p5
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/louis/Documents/Code/p5.js/npm-debug.log
```
[npm-debug.log](https://github.com/processing/p5.js/files/3184424/npm-debug.log)

**Comment by Spongman** [comment]
Try doing an `npm install` first.

**Comment by Nekzuris** [comment]
I did, thought there where a few warnings but no error.

**Comment by Nekzuris** [comment]
I've re-clone the repository and when I run `npm install` I get these warning:

[log.txt](https://github.com/processing/p5.js/files/3184465/log.txt)

**Comment by stalgiag** [comment]
Hm I am running a more recent node/npm setup but I don't know if that would be it. Updating might be worth a try if you are in a position to do so, otherwise I can investigate a bit later. Also we can wait to see if anyone has had a similar problem before.

**Comment by Nekzuris** [comment]
Yes! it work with node v10.15 and npm v6.4.1 but I've got this after `npm install`
```bash
added 1323 packages from 1487 contributors and audited 7120 packages in 61.467s
found 30 vulnerabilities (3 low, 18 moderate, 9 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

**Comment by Spongman** [comment]
that's fine, you can ignore those.

**Comment by Nekzuris** [comment]
So I've made the changes, run grunt, commit but I'm unable tu push, did you know why?

```bash
[00:46:58]louis@Lenovo-ideapad:p5.js$ ssh -T git@github.com
Hi Nekzuris! You've successfully authenticated, but GitHub does not provide shell access.
[00:48:05]louis@Lenovo-ideapad:p5.js$ git push origin fix-nfs
Username for 'https://github.com': Nekzuris
Password for 'https://Nekzuris@github.com': 
remote: Permission to processing/p5.js.git denied to Nekzuris.
fatal: unable to access 'https://github.com/processing/p5.js.git/': The requested URL returned error: 403
```

**Comment by stalgiag** [comment]
Maybe you aren't working off of your fork? I am not sure but it looks like you are somehow trying to push to origin which is `processing/p5.js` but you need to push to `nekzuris/p5.js`

**Comment by Nekzuris** [comment]
Ok so it looks like I've made my first pull request :smiley:

Thanks you @stalgiag and @Spongman :+1:

**Comment by stalgiag** [comment]
closed via #3747
